crates/services/block_aggregator_api/src/api/protobuf_adapter.rs

```diff
@@ -182,4 +194,10 @@ impl BlockAggregatorApi for ProtobufAPI {
     }
 }
 
-pub struct ProtobufClient;
+impl Drop for ProtobufAPI {
+    fn drop(&mut self) {
+        if let Some(shutdown_sender) = self.shutdown_sender.take() {
+            let _ = shutdown_sender.send(());
+        }
+    }
+}
```

Comment:
This is better than nothing, but if it isn't dropped properly it will be orphaned still. Might be fine.